Loop/Managers/LoopDataManager.swift

```diff
@@ -10,150 +10,158 @@ import Foundation
 import Combine
 import HealthKit
 import LoopKit
+import LoopKitUI
 import LoopCore
 import WidgetKit
 
-protocol PresetActivationObserver: AnyObject {
-    func presetActivated(context: TemporaryScheduleOverride.Context, duration: TemporaryScheduleOverride.Duration)
-    func presetDeactivated(context: TemporaryScheduleOverride.Context)
+struct AlgorithmDisplayState {
+    var input: LoopAlgorithmInput?
+    var output: LoopAlgorithmOutput?
+
+    var activeInsulin: InsulinValue? {
+        guard let input, let value = output?.activeInsulin else {
+            return nil
+        }
+        return InsulinValue(startDate: input.predictionStart, value: value)
+    }
+
+    var activeCarbs: CarbValue? {
+        guard let input, let value = output?.activeCarbs else {
+            return nil
+        }
+        return CarbValue(startDate: input.predictionStart, value: value)
+    }
+
+    var asTuple: (algoInput: LoopAlgorithmInput?, algoOutput: LoopAlgorithmOutput?) {
+        return (algoInput: input, algoOutput: output)
+    }
+}
+
+protocol DeliveryDelegate: AnyObject {
+    var isSuspended: Bool { get }
+    var pumpInsulinType: InsulinType? { get }
+    var basalDeliveryState: PumpManagerStatus.BasalDeliveryState? { get }
+    var isPumpConfigured: Bool { get }
+
+    func enact(_ recommendation: AutomaticDoseRecommendation) async throws
+    func enactBolus(units: Double, activationType: BolusActivationType) async throws
+    func roundBasalRate(unitsPerHour: Double) -> Double
+    func roundBolusVolume(units: Double) -> Double
+}
+
+protocol DosingManagerDelegate {
+    func didMakeDosingDecision(_ decision: StoredDosingDecision)
+}
+
+enum LoopUpdateContext: Int {
+    case insulin
+    case carbs
+    case glucose
+    case preferences
+    case forecast
 }
 
+@MainActor
 final class LoopDataManager {
-    enum LoopUpdateContext: Int {
-        case insulin
-        case carbs
-        case glucose
-        case preferences
-        case loopFinished
-    }
+    nonisolated static let LoopUpdateContextKey = "com.loudnate.Loop.LoopDataManager.LoopUpdateContext"
 
-    let loopLock = UnfairLock()
+    // Represents the current state of the loop algorithm for display
+    var displayState = AlgorithmDisplayState()
```

Comment:
This represents the most recent run of the Loop algorithm, and data from here is what is shown to the user on the main status page.